stdlib/public/core/EmbeddedStubs.swift

```diff
@@ -0,0 +1,303 @@
+}
+
+@_unavailableInEmbedded
+public protocol TextOutputStreamable {
+  func write<Target: TextOutputStream>(to target: inout Target)
+}
+
+@_unavailableInEmbedded
+extension String {
+  internal static func _fromUTF8Repairing(_ input: UnsafeBufferPointer<UInt8>) -> (result: String, repairsMade: Bool) { fatalError() }
+}
+
+@_unavailableInEmbedded
+extension String {
+  internal static func _fromASCII(_ input: UnsafeBufferPointer<UInt8>) -> String { fatalError() }
+  internal static func _uncheckedFromUTF8( _ input: UnsafeBufferPointer<UInt8>) -> String { fatalError() }
+  internal static func _uncheckedFromUTF8(_ input: UnsafeBufferPointer<UInt8>, isASCII: Bool) -> String { fatalError() }
+  internal static func _uncheckedFromUTF8(_ input: UnsafeBufferPointer<UInt8>, asciiPreScanResult: Bool) -> String { fatalError() }
+}
+
+@_unavailableInEmbedded
+extension String {
+  public init<T: BinaryInteger>(_ value: T, radix: Int = 10, uppercase: Bool = false) { fatalError() }
+}
+
+/// Unicode.Scalar
+
+public enum Unicode {}
+
+extension Unicode {
+  public struct Scalar: Sendable {
+    internal var _value: UInt32
+    internal init(_value: UInt32) {
+      self._value = _value
+    }
+  }
+}
+
+extension Unicode.Scalar : _ExpressibleByBuiltinUnicodeScalarLiteral, ExpressibleByUnicodeScalarLiteral {
+  public var value: UInt32 { return _value }
+  public init(_builtinUnicodeScalarLiteral value: Builtin.Int32) {
+    self._value = UInt32(value)
+  }
+  public init(unicodeScalarLiteral value: Unicode.Scalar) {
+    self = value
+  }
+  public init?(_ v: UInt32) {
+    if (v < 0xD800 || v > 0xDFFF) && v <= 0x10FFFF {
+      self._value = v
+      return
+    }
+    return nil
+  }
+  public init?(_ v: UInt16) {
+    self.init(UInt32(v))
+  }
+  public init(_ v: UInt8) {
+    self._value = UInt32(v)
+  }
+  public init(_ v: Unicode.Scalar) {
+    self = v
+  }
+  @_unavailableInEmbedded
+  public func escaped(asASCII forceASCII: Bool) -> String { fatalError() }
+  @_unavailableInEmbedded
+  internal func _escaped(asASCII forceASCII: Bool) -> String? { fatalError() }
+  public var isASCII: Bool {
+    return value <= 127
+  }
+  internal var _isPrintableASCII: Bool {
+    return (self >= Unicode.Scalar(0o040) && self <= Unicode.Scalar(0o176))
+  }
+}
+
+extension Unicode.Scalar: Equatable {
+  public static func == (lhs: Unicode.Scalar, rhs: Unicode.Scalar) -> Bool {
+    return lhs.value == rhs.value
+  }
+}
+
+extension Unicode.Scalar: Comparable {
+  public static func < (lhs: Unicode.Scalar, rhs: Unicode.Scalar) -> Bool {
+    return lhs.value < rhs.value
+  }
+}
+
+public typealias UTF8 = Unicode.UTF8
+
+extension Unicode {
+  public enum UTF8 {
+  }
+}
+
+extension Unicode.UTF8 {
+  public typealias CodeUnit = UInt8
+}
+
+/// Codable
+
+@_unavailableInEmbedded
+public protocol Encodable {
+  func encode(to encoder: any Encoder) throws
+}
+
+@_unavailableInEmbedded
+public protocol Decodable {
+  init(from decoder: any Decoder) throws
```

Comment:
if we are removing existentials we need to consider `Codable` since that is a requirement for that protocol. Altering the requirements to be `some Decoder` would violate source sharing between embedded and normal swift modes.

This would likely mean that we should consider a potential replacement for both embedded systems and normal systems that avoids existentials (but I don't think you can fully avoid heap allocations via reference types)